tests/e2e/vmc_vc_cert_rotate.go

```diff
@@ -0,0 +1,217 @@
+	f := framework.NewDefaultFramework("vmc-upgrade-tkg")
+	var (
+		client            clientset.Interface
+		namespace         string
+		svNamespace       string
+		scParameters      map[string]string
+		storagePolicyName string
+		manifestPath      = "tests/e2e/testing-manifests/statefulset/nginx"
+		refreshToken      string
+		orgID             string
+		sddcID            string
+	)
+
+	ginkgo.BeforeEach(func() {
+		client = f.ClientSet
+		namespace = getNamespaceToRunTests(f)
+		svcClient, svNamespace = getSvcClientAndNamespace()
+		nodeList, err := fnodes.GetReadySchedulableNodes(f.ClientSet)
+		framework.ExpectNoError(err, "Unable to find ready and schedulable Node")
+		storagePolicyName = GetAndExpectStringEnvVar(envStoragePolicyNameForSharedDatastores)
+		if !(len(nodeList.Items) > 0) {
+			framework.Failf("Unable to find ready and schedulable Node")
+		}
+		bootstrap()
+		setResourceQuota(svcClient, svNamespace, rqLimit)
+		scParameters = make(map[string]string)
+		storagePolicyName = GetAndExpectStringEnvVar(envStoragePolicyNameForSharedDatastores)
+	})
+	ginkgo.AfterEach(func() {
+		svcClient, svNamespace := getSvcClientAndNamespace()
+		setResourceQuota(svcClient, svNamespace, defaultrqLimit)
+	})
+
+	/*
+		Test to validate the tkg cluster post vc cert rotation in vmc env
+		Steps
+			1. Create statefulset pods with 3 replicas
+			2. Execute the RTS script to perform cert rotation
+			3. Scale Up the statefulset pods and verify volumes are attached to the node
+			5. Scale Down the statefulset pods and verify volumes are detached from the node
+			6. Delete the statefulset pods and PVC's
+	*/
+
+	ginkgo.It("[vmc] VC Cert Rotate in VMC", func() {
+		var sc *storagev1.StorageClass
+		var err error
+		ctx, cancel := context.WithCancel(context.Background())
+		defer cancel()
+
+		refreshToken = os.Getenv("REFRESH_TOKEN")
+		if refreshToken == "" {
+			ginkgo.Skip("Env REFRESH_TOKEN is missing")
+		}
+		orgID = os.Getenv("ORG_ID")
+		if orgID == "" {
+			ginkgo.Skip("Env ORG_ID is missing")
+		}
+		sddcID = os.Getenv("SDDC_ID")
+		if orgID == "" {
+			ginkgo.Skip("Env SDDC_ID is missing")
+		}
+
+		ginkgo.By("Creating StorageClass for Statefulset")
+		scParameters[svStorageClassName] = storagePolicyName
+		scSpec := getVSphereStorageClassSpec(defaultNginxStorageClassName, scParameters, nil, "", "", false)
+		sc, err = client.StorageV1().StorageClasses().Create(ctx, scSpec, metav1.CreateOptions{})
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		defer func() {
+			err := client.StorageV1().StorageClasses().Delete(ctx, sc.Name, *metav1.NewDeleteOptions(0))
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		}()
+
+		ginkgo.By("Creating statefulset")
+		statefulset := fss.CreateStatefulSet(client, manifestPath, namespace)
+		defer func() {
+			ginkgo.By(fmt.Sprintf("Deleting all statefulsets in namespace: %v", namespace))
+			fss.DeleteAllStatefulSets(client, namespace)
+			if supervisorCluster {
```

Comment:
why we are not deleting SC  for other testbed types?

Reply:
This check is not required since in VMC we support only GC, hence remove this check